src/Symfony/Component/Messenger/Transport/SqsExt/Connection.php

```diff
@@ -0,0 +1,364 @@
+    {
+        if (null === $this->currentResponse) {
+            $this->currentResponse = $this->request($this->getQueueUrl(), [
+                'Action' => 'ReceiveMessage',
+                'VisibilityTimeout' => $this->configuration['visibility_timeout'],
+                'MaxNumberOfMessages' => $this->configuration['buffer_size'],
+                'WaitTimeSeconds' => $this->configuration['wait_time'],
+            ]);
+        }
+
+        foreach ($this->client->stream($this->currentResponse) as $chunk) {
+            if (!$chunk->isLast()) {
+                return;
+            }
+        }
+
+        $xml = new \SimpleXMLElement($this->currentResponse->getContent());
+        foreach ($xml->ReceiveMessageResult->Message as $xmlMessage) {
+            $this->buffer[] = [
+                'id' => (string) $xmlMessage->ReceiptHandle,
+            ] + json_decode($xmlMessage->Body, true);
+        }
+
+        $this->currentResponse = null;
+
+        yield from $this->getPendingMessages();
+    }
+
+    public function setup(): void
+    {
+        $this->call($this->configuration['endpoint'], [
+            'Action' => 'CreateQueue',
+            'QueueName' => $this->configuration['queue_name'],
+        ]);
+        $this->queueUrl = null;
+
+        $this->configuration['auto_setup'] = false;
+    }
+
+    public function delete(string $id): void
+    {
+        $this->call($this->getQueueUrl(), [
+            'Action' => 'DeleteMessage',
+            'ReceiptHandle' => $id,
+        ]);
+    }
+
+    public function getMessageCount(): int
+    {
+        $response = $this->request($this->getQueueUrl(), [
+            'Action' => 'GetQueueAttributes',
+            'AttributeNames' => ['ApproximateNumberOfMessages'],
+        ]);
+        $this->checkResponse($response);
+        $xml = new \SimpleXMLElement($response->getContent());
+        foreach ($xml->GetQueueAttributesResult->Attribute as $attribute) {
+            if ('ApproximateNumberOfMessages' !== (string) $attribute->Name) {
+                continue;
+            }
+
+            return (int) $attribute->Value;
+        }
+
+        return 0;
+    }
+
+    public function requeue(string $id, int $delay = 0): void
+    {
+        $this->call($this->getQueueUrl(), [
+            'Action' => 'ChangeMessageVisibility',
+            'ReceiptHandle' => $id,
+            'VisibilityTimeout' => $delay,
+        ]);
+    }
+
+    public function requeuePrefetchedMessages(): void
+    {
+        if (null !== $this->currentResponse) {
+            $this->currentResponse->cancel();
+        }
+
+        foreach ($this->getPendingMessages() as $message) {
+            $this->requeue($message['id']);
+        }
+    }
+
+    public function send(string $body, array $headers, int $delay = 0): void
+    {
+        if ($this->configuration['auto_setup']) {
+            $this->setup();
+        }
+
+        $this->call($this->getQueueUrl(), [
```

Comment:
I see that you miss support for FIFO queues. To add a message to a FIFO queue you need a `MessageGroupId`. (That parameter cannot be present when sending to a "standard queue".

Reply:
I had suggested to implement such feature in dedicated PR https://github.com/symfony/symfony/pull/32454#issuecomment-519878880